packages/react-next/src/components/MessageBar/MessageBar.base.tsx

```diff
@@ -1,210 +1,138 @@
 import { IconButton } from '../../compat/Button';
 import { Icon } from '../../Icon';
 import { IMessageBarProps, IMessageBarStyleProps, IMessageBarStyles, MessageBarType } from './MessageBar.types';
+import { useId, useBoolean } from '@uifabric/react-hooks';
 
 const getClassNames = classNamesFunction<IMessageBarStyleProps, IMessageBarStyles>();
 
-export interface IMessageBarState {
-  labelId?: string;
-  showContent?: boolean;
-  expandSingleLine?: boolean;
-}
-
-export class MessageBarBase extends React.Component<IMessageBarProps, IMessageBarState> {
-  public static defaultProps: IMessageBarProps = {
-    messageBarType: MessageBarType.info,
-    onDismiss: undefined,
-    isMultiline: true,
-  };
-
-  private ICON_MAP = {
-    [MessageBarType.info]: 'Info',
-    [MessageBarType.warning]: 'Info',
-    [MessageBarType.error]: 'ErrorBadge',
-    [MessageBarType.blocked]: 'Blocked2',
-    [MessageBarType.severeWarning]: 'Warning',
-    [MessageBarType.success]: 'Completed',
-  };
-
-  private _classNames: { [key in keyof IMessageBarStyles]: string };
+const COMPONENT_NAME = 'MessageBar';
+
+const ICON_MAP = {
+  [MessageBarType.info]: 'Info',
+  [MessageBarType.warning]: 'Info',
+  [MessageBarType.error]: 'ErrorBadge',
+  [MessageBarType.blocked]: 'Blocked2',
+  [MessageBarType.severeWarning]: 'Warning',
+  [MessageBarType.success]: 'Completed',
+};
+
+export const MessageBarBase = React.forwardRef<HTMLDivElement, IMessageBarProps>((props, ref) => {
+  const [expandSingleLine, { toggle: toggleExpandSingleLine }] = useBoolean(false);
+  const labelId = useId('MessageBar');
+
+  const {
+    actions,
+    className,
+    children,
+    overflowButtonAriaLabel,
+    dismissIconProps,
+    styles,
+    theme,
+    messageBarType = MessageBarType.info,
+    onDismiss = undefined,
+    isMultiline = true,
+    truncated,
+    dismissButtonAriaLabel,
+    messageBarIconProps,
+  } = props;
+
+  const nativeProps = getNativeProps<React.HTMLAttributes<HTMLSpanElement>>(props, htmlElementProperties, [
+    'className',
+  ]);
+
+  const classNames: { [key in keyof IMessageBarStyles]: string } = getClassNames(styles, {
+    theme: theme!,
+    messageBarType: messageBarType || MessageBarType.info,
+    onDismiss: onDismiss !== undefined,
+    actions: actions !== undefined,
+    truncated: truncated,
+    isMultiline: isMultiline,
+    expandSingleLine: expandSingleLine,
+    className,
+  });
+
+  const iconProps = { iconName: expandSingleLine ? 'DoubleChevronUp' : 'DoubleChevronDown' };
+
+  const getActionsDiv = React.useCallback((): JSX.Element | null => {
+    if (actions) {
+      return <div className={classNames.actions}>{actions}</div>;
+    }
+    return null;
+  }, [actions, classNames.actions]);
 
-  constructor(props: IMessageBarProps) {
-    super(props);
+  const getAnnouncementPriority = React.useCallback((): 'assertive' | 'polite' => {
```

Comment:
I'd recommend moving `getAnnouncementPriority` outside the component body and adding a `messageBarType` arg, since this is just a helper not a callback.